### PR TITLE
Convert colossal oscar da

### PR DESCRIPTION
**Pull request by peterbjorgensen**
Add colossal oscar dataset (da) part. The script can easily be extended to all the other languages.

### CODE REVIEW

File: data-processing/scripts/convert_colossal_oscar_10_to_jsonlgz.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                output_path = output_dir / ("__".join(rel_input_path_parts[:-1]) + "__" + 
+                    Path(Path(rel_input_path_parts[-1]).stem).stem + f"_{doc_count//DOCS_PER_FILE}.jsonl.gz"
+                )
+                if out_fh is not None:
+                    out_fh.close()
+                out_fh = gzip.open(output_path, "wt")
+            obj = json.loads(line)
+            new_obj: dict[str, Union[dict[str, Any], str]] = {
+                "id": str(obj["warc_headers"]["warc-record-id"]), # Copy metadata id to root
+                "text": obj["content"],
```
Comment (KennethEnevoldsen) [inline data-processing/scripts/convert_colossal_oscar_10_to_jsonlgz.py:94]: Is content cleaned from HTML?

File: data-processing/scripts/convert_colossal_oscar_10_to_jsonlgz.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    assert (
+        len(sys.argv) == 3
+    ), "Usage: convert_colossal_oscar_10_to_jsonlgz.py colossal_oscar_root output_directory"
+
+    oscar_root_dir = Path(sys.argv[1])
+    output_dir = Path(sys.argv[-1])
+
+    lang_codes = ["da", "sv", "no", "nn", "is"]
+    for lang in lang_codes:
+        input_files = oscar_root_dir.glob(f"./**/{lang}_meta/*.jsonl.zst")
```
Comment (KennethEnevoldsen) [inline data-processing/scripts/convert_colossal_oscar_10_to_jsonlgz.py:118]: zst? Shouldn't we keep it all as json.gz?

**Review by KennethEnevoldsen — COMMENTED** [review]
I assume there is a smart reason for choosing the format, but it would be nice to know.

@jankounchained you might be interested in this as well.

File: data-processing/scripts/convert_colossal_oscar_10_to_jsonlgz.py (hunk 1 above)
Comment (peterbjorgensen) [reply data-processing/scripts/convert_colossal_oscar_10_to_jsonlgz.py:94]: They use ungoliant which uses the WET files that are roughly cleaned from HTML, but it's probably full of noise. https://huggingface.co/datasets/oscar-corpus/colossal-oscar-1.0/blob/main/README.md

File: data-processing/scripts/convert_colossal_oscar_10_to_jsonlgz.py (hunk 2 above)
Comment (peterbjorgensen) [reply data-processing/scripts/convert_colossal_oscar_10_to_jsonlgz.py:118]: zst is only for the input (raw) dataset files. The output of the script is saved in jsonl.gz

**Review by KennethEnevoldsen — APPROVED** [review]

**Pull request merged**